pkg/function/server/server.go

```diff
@@ -90,39 +90,52 @@ func (s *server) Start(ctx context.Context, inputOptions ...Option) {
 		inputOption(opts)
 	}
 
-	cleanup := func() {
-		if _, err := os.Stat(opts.sockAddr); err == nil {
-			if err := os.RemoveAll(opts.sockAddr); err != nil {
-				log.Fatal(err)
-			}
+	cleanup := func() error {
+		_, err := os.Stat(opts.sockAddr)
+		if err == nil {
+			err = os.RemoveAll(opts.sockAddr)
 		}
+		return err
+	}
+
+	err := cleanup()
+	if err != nil {
+		return err
 	}
-	cleanup()
 
 	ctxWithSignal, stop := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
 	defer stop()
 
 	lis, err := net.Listen(functionsdk.Protocol, opts.sockAddr)
 	if err != nil {
-		log.Fatalf("failed to execute net.Listen(%q, %q): %v", functionsdk.Protocol, functionsdk.Addr, err)
+		return err
 	}
 	grpcServer := grpc.NewServer(
 		grpc.MaxRecvMsgSize(opts.maxMessageSize),
 		grpc.MaxSendMsgSize(opts.maxMessageSize),
 	)
 	functionpb.RegisterUserDefinedFunctionServer(grpcServer, s.svc)
 
+	errCh := make(chan error, 1)
```

Comment:
can you close the errCh? perhaps, you can do a `defer` on closing the channel?

Reply:
done